### PR TITLE
feat(esl-utils): add `parseTime`, less restrictive alternative to passed time format

### PR DESCRIPTION
**Pull request by ala-n**

### CODE REVIEW

File: src/modules/esl-utils/misc/format.ts [hunk 1]
```diff
@@ -90,23 +90,32 @@ export function format(str: string, source: Record<string, any>, matcher: RegExp
...
-  if (str.endsWith('ms')) return +str.slice(0, -2);
-  return +str.slice(0, -1) * 1000;
+  const parseNoEmpty = (str: string): number => str ? +str : NaN;
+  if (str.endsWith('ms')) return parseNoEmpty(str.slice(0, -2));
+  if (str.endsWith('s')) return parseNoEmpty(str.slice(0, -1)) * 1000;
+  return +str; // empty string without unit is treated as 0
 }
 
+/** @deprecated alias for {@link parseTime} */
+export const parseCSSTime = parseTime;
```
Comment (ala-n) [inline src/modules/esl-utils/misc/format.ts:107]: Optional: 
```suggestion
const CSS_TIME_FORMAT = /(\d*\.?\d+)(ms|s)/;

/**
 * Restrictive time parser according to [CSS style](https://developer.mozilla.org/en-US/docs/Web/CSS/time) spec.
 * @see {@link parseTime}
 */
export const parseCSSTime = (timeStr: string): number =>
  CSS_TIME_FORMAT.test(timeStr) ? parseTime(timeStr) : NaN;
```

File: src/modules/esl-utils/misc/format.ts [hunk 2]
```diff
@@ -90,23 +90,32 @@ export function format(str: string, source: Record<string, any>, matcher: RegExp
...
  * @returns number[] - array of times in milliseconds
 */
-export function parseCSSTimeSet(timeStr: string): number[] {
-  return timeStr.split(',').map((timeSubstr) => parseCSSTime(timeSubstr));
+export function parseTimeSet(timeStr: string): number[] {
+  return timeStr.split(',').map((timeSubstr) => parseTime(timeSubstr));
 }
+
+/** @deprecated alias for {@link parseTimeSet} */
+export const parseCSSTimeSet = parseTimeSet;
```
Comment (ala-n) [inline src/modules/esl-utils/misc/format.ts:121]: Optional: 
```suggestion
/**
 * Restrictive timeSet parser according to [CSS style](https://developer.mozilla.org/en-US/docs/Web/CSS/time) spec.
 * @see {@link parseCSSTime}
 */
export function parseCSSTimeSet(timeStr: string): number[] {
  return timeStr.split(',').map((timeSubstr) => parseCSSTime(timeSubstr));
}
```

**Review by abarmina — APPROVED** [review]

File: src/modules/esl-utils/misc/format.ts [hunk 3]
```diff
@@ -90,23 +90,32 @@ export function format(str: string, source: Record<string, any>, matcher: RegExp
...
 */
-export function parseCSSTime(timeStr: string): number {
+export function parseTime(timeStr: string): number {
   const str = timeStr.trim().toLowerCase();
-  if (!/\dm?s$/.test(str)) return NaN;
-  if (str.endsWith('ms')) return +str.slice(0, -2);
-  return +str.slice(0, -1) * 1000;
+  const parseNoEmpty = (s: string): number => s ? +s : NaN;
+  if (str.endsWith('ms')) return parseNoEmpty(str.slice(0, -2));
+  if (str.endsWith('s')) return parseNoEmpty(str.slice(0, -1)) * 1000;
```
Comment (dshovchko) [inline src/modules/esl-utils/misc/format.ts:102]: Should we parse invalid values like "12 s" or "200 ms"?

**Review by dshovchko — APPROVED** [review]

**Review by dshovchko — APPROVED** [review]

**Pull request merged**